### PR TITLE
profiler for example selective-tp

### PR DESCRIPTION
**Pull request by moonbucks**
## Type of change

profiler added

## Checklist:

- [v] Has code been commented, particularly in hard-to-understand areas?

### CODE REVIEW

File: examples/selective2d/2d_train.py [hunk 1]
```diff
@@ -328,21 +331,30 @@ def pp_tp_train(stage, mesh, args):
...
+        ),
+    ) as prof:
+        while local_iter_num < train_iters:
+            optimizer.zero_grad()
+            t0 = time.perf_counter()
+            X, Y = get_rand(args)
+            if pp_rank == 0:
+                out = stage(X)
+            elif pp_rank == args.pp_size - 1:
+                out = stage(Y)
```
Comment (fduwjj) [inline examples/selective2d/2d_train.py:347]: does this mean we have two pipeline here?

**Review by fduwjj — APPROVED** [review]
LGTM

**Comment by fduwjj** [comment]
Let's also make sure Lint is happy before merge, thanks!

File: examples/selective2d/2d_train.py (hunk 1 above)
Comment (moonbucks) [reply examples/selective2d/2d_train.py:347]: we have args.pp_size stages but we only pass 'input' to the first stage and 'label' to the last stage (to calculate loss and backpropagation). intermediate values are automatically passed by pippy
